### PR TITLE
Fixes for DB index

### PR DESCRIPTION
**Pull request by mboudet**
db index was semi-functional in Tripal 3 (due to issues in tripal_rest_api)

Added 'node' mode -> Index website nodes (mostly useless in tripal 3, unless legacy modules maybe)
Added 'entities' mode (for Tripal 3 only) -> Index website entities

'website' mode will redirect to 'entities' in tripal 3, and 'nodes' in tripal 2. (What the user expect to index)

Added db tune, which is required for db index --mode 'entities' 

Added optional parameters for tripal 3

### CODE REVIEW

**Comment by mboudet** [comment]
It should probably good to add a separate route for updating an existing index

File: tripal/db/__init__.py [hunk 1]
```diff
@@ -122,11 +134,35 @@ def index(self, mode="website", table=None, index_name=None, queues=10, fields=[
...
+
+        # 'website' redirect to node indexing in tripal code, but it's misleading.
+        # We redirect users to entity indexing if Tripal 3, and node indexing in tripal 2
+        # We leave the mode for node indexing at 'website' in Tripal 2 for back-compatibility
+
+        if self.tripal.version == 3:
+            if mode == 'website':
+                mode = 'entities'
+        else:
+            if mode == 'nodes':
```
Comment (abretaud) [inline tripal/db/__init__.py]: I guess it should be website on this line and nodes on the next one

Comment (abretaud) [reply tripal/db/__init__.py]: nevermind

**Comment by abretaud** [comment]
Ready to merge or should we wait for #10 ?

**Comment by mboudet** [comment]
We can merge I think. It was non-functional before, and it will still be until the PR merge, so there's no loss.

**Pull request merged**